04-tracking.md

```diff
@@ -0,0 +1,145 @@
+---
+layout: page
+title: Version Control with Mercurial
+subtitle: Tracking Changes to Files
+---
+> ## Learning Objectives {.objectives}
+>
+> * Display the version control status of files in a repository and explain what those statuses mean.
+> * Add files to Mercurial's collection of tracked files.
+> * Record metadata about changes to a file.
+> * Display the history of changes to files in a repository and explain the metadata that is recorded with each changeset.
+
+Let's create a file called `plan.txt` in which Susan is going to write her
+initial ideas and notes about the Salish Sea NEMO daily forecast system.
+You can use any text editor you want
+(`Notepad`, `TextEdit`, `gedit`, `nano`, `emacs`, `vi`, ...)
+but please don't use a word processor like Microsoft Word or LibreOffice Write.
+In particular,
+the editor does not have to be the one that you set in your Mercurial
+configuration earlier.)
+In what follows it is assumed that `nano` is used.
+
+~~~ {.bash}
+$ nano plan.txt
+~~~
+
+and type into that file:
+
+~~~ {.output}
+Goal: Run NEMO everyday to forecast storm surge water levels
+~~~
+
+`plan.txt` has now been created and it contains a single line:
+
+~~~ {.bash}
+$ ls
+~~~
+~~~ {.output}
+plan.txt
+~~~
+~~~ {.bash}
+$ cat plan.txt
+~~~
+~~~ {.output}
+Goal: Run NEMO everyday to forecast storm surge water levels
+~~~
+
+We can ask Mercurial to tell us what it knows about the files in our project with the `hg status` command.
+Mercurial tells us that it has noticed the new file:
+
+~~~ {.bash}
+$ hg status
+~~~
+~~~ {.output}
+? plan.txt
+~~~
+
+The `?` at the beginning of the line means that Mercurial isn't keeping track of the file.
+We can tell Mercurial that it should do so using `hg add`:
+
+~~~ {.bash}
+$ hg add plan.txt
+~~~
+
+and then check that the right thing happened:
+
+~~~ {.bash}
+$ hg status
+~~~
+~~~ {.output}
+A plan.txt
+~~~
+
+Mercurial now knows that it's supposed to keep track of `plan.txt`,
+but it hasn't yet recorded any changes for posterity as a commit.
+To get it to do that,
+we need to run one more command:
+
+~~~ {.bash}
+$ hg commit -m "Starting to plan the daily NEMO forecast system."
+~~~
+
+When we run `hg commit`,
+Mercurial takes the file we have told it about by using `hg add` and stores a copy permanently inside the special `.hg` directory.
+That permanent copy is called a [commit](reference.html#commit)
+(or [revision](reference.html#revision)).
+
+We use the `-m` flag (for "message") to record a comment that will help us remember later on what we did and why.
+If we just run `hg commit` without the `-m` option,
+Mercurial will launch `nano`
+(or whatever other editor we configured at the start)
+so that we can write a longer message.
+
+[Good commit messages][commit-messages] start with a brief (<70 characters) summary of
```

Comment:
Credit to the git-novice lesson for reminding me of that link in this round of edits, though I know I've liked to it in other material.

On the topic of reference links, I like to include a richer collection on the index page than the generic SWC lesson does (Reference, Discussion, Instructor's Guide). I usually add the Mercurial site, the Mercurial - The Definitive Guide book site, and a collection of "[command reminders](https://douglatornell.github.io/hg-novice-2015-09-22-ubc/commands.html)". Your comment makes me think that the Good Commit Messages link would be a good addition to that.

What do you think about making that collection of links a standard thing on the index page for the Mercurial lesson?